extensions/smallrye-reactive-messaging/deployment/src/test/java/io/quarkus/smallrye/reactivemessaging/signatures/IncomingsTest.java

```diff
@@ -0,0 +1,75 @@
+package io.quarkus.smallrye.reactivemessaging.signatures;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.awaitility.Awaitility.await;
+
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+import org.eclipse.microprofile.reactive.messaging.Incoming;
+import org.eclipse.microprofile.reactive.messaging.Outgoing;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.reactivestreams.Publisher;
+
+import io.quarkus.test.QuarkusUnitTest;
+import io.smallrye.mutiny.Multi;
+
+public class IncomingsTest {
+
+    @RegisterExtension
+    static final QuarkusUnitTest config = new QuarkusUnitTest()
+            .setArchiveProducer(() -> ShrinkWrap.create(JavaArchive.class)
+                    .addClasses(ProducerOnA.class, ProducerOnB.class, MyBeanUsingMultipleIncomings.class));
+
+    @Inject
+    MyBeanUsingMultipleIncomings bean;
+
+    @Test
+    public void testIncomingsWithTwoSources() {
+        await().until(() -> bean.list().size() == 6);
+        assertThat(bean.list()).containsExactlyInAnyOrder("a", "b", "c", "d", "e", "f");
```

Comment:
In addition to this line, I'd also add

```java
assertThat(bean.list()).containsSubsequence("a", "b", "c");
assertThat(bean.list()).containsSubsequence("d", "e", "f");
```

to verify ordering.

Reply:
Good idea.